packages/flutter_tools/lib/src/android/android_device_discovery.dart

```diff
@@ -154,6 +162,12 @@ class AndroidDevices extends PollingDeviceDiscovery {
             productID: info['product'],
             modelID: info['model'] ?? deviceID,
             deviceCodeName: info['device'],
+            androidSdk: androidSdk ?? globals.androidSdk,
```

Comment:
I don't love this, but I also didn't want to change absolute everything.

Medium term, we should split up the device object from the device service.